src/components/FilterTags/FilterTags.js

```diff
@@ -87,20 +88,95 @@ const FilterTags = () => {
     };
   }, []);
 
-  const handleFilterClick = () => {
-    console.log("Filter clicked");
+  /**
+  Apply a given filter
+  @param {object} e The event object
+  @param {string} filter The name of the filter
+   @param {string} filterBy The name of the filter group
+*/
+  const addActiveFilter = (e, filter, filterBy) => {
+    setActiveFilters(filters => {
+      const updatedFilters = { ...filters };
+      updatedFilters[filterBy] = updatedFilters[filterBy] || [];
+      if (!updatedFilters[filterBy].includes(filter)) {
+        updatedFilters[filterBy].push(filter);
+      }
+      return updatedFilters;
+    });
+    e.currentTarget.classList.add("is-selected");
   };
 
+  /**
+  Remove a given filter
+  @param {object} e The event object
+  @param {string} filter The name of the filter
+   @param {string} filterBy The name of the filter group
+*/
+  const removeActiveFilter = (e, filter, filterBy) => {
```

Comment:
if the filter value contained the group and the value then it would simplify this logic because you could just push them into a set/array and then extract the group/value when rendering or parsing.

For example 
```js
const filters = new Set(['cloud:google', 'credential:secret', ...]);
``` 

A set guarantees unique entries and it's iterable so you can loop through the set to generate the elements and set selections to it without worrying about duplicates.